### PR TITLE
AGSEC-201/203: Move the OAuth2 configuration to aerogear.org

### PR DESCRIPTION
**Pull request by abstractj**
This is just the start from the series of section for security, I prefer to deliver early and often instead of a gazillion of documentation to review.

To track all the items being created check [AGSEC-200](https://issues.jboss.org/browse/AGSEC-200).

@danielpassos @matzew @lfryc @cvasilak @secondsun @edewit @corinnekrych you are all invited to review.

Keep in mind that the content will be moved gradually to the `new-design` branch and I also made sure to keep the ownership of the document.

### CODE REVIEW

**Comment by edewit** [comment]
The google description you have here only works for Android. If you take the one from [here](https://github.com/aerogear/aerogear-cordova-cookbook/tree/master/Shoot), that will work for all platforms Android, iOS and Windows.

**Comment by corinnekrych** [comment]
Google working with iOS is a bit more tricky, @edewit doc is good.
+1 on moving this doc, I also think this ticket should also include cleaning the README instruction (pointing to ag.org instead). Too much info:
https://github.com/aerogear/aerogear-ios-cookbook/tree/master/Shoot#facebook-setup
https://github.com/aerogear/aerogear-android-cookbook/tree/master/ShootAndShare#google-account-setup
and cordova and windows.

**Comment by matzew** [comment]
> I also think this ticket should also include cleaning the README instruction (pointing to ag.org instead). Too much info

@corinnekrych mind filing a separate JIRA / PR for this ?

File: docs/guides/index.markdown [hunk 1]
```diff
@@ -3,22 +3,36 @@ layout: basic
...
+
+* OAuth2
+  1. [Basic steps](security/oauth2-basic-steps)
+  2. Client libraries
+
+     The following SDKs were crafted with the simplicity in mind and are constantly improving at each release.
+
+     * [Android client library](http://search.maven.org/#artifactdetails%7Corg.jboss.aerogear%7Caerogear-android-authz%7C2.0.0%7Caar)
+     * [iOS client library](http://cocoapods.org/?q=aerogearoauth2)
+     * [Apache Cordova client library](https://github.com/aerogear/aerogear-oauth2-cordova/releases/tag/v1.0.2)
```
Comment (matzew) [inline docs/guides/index.markdown]: Let's use this link: http://plugins.cordova.io/#/package/org.jboss.aerogear.cordova.oauth2

**Comment by matzew** [comment]
> The google description you have here only works for Android. If you take the one from here, that will work for all platforms Android, iOS and Windows.

Sounds good using that info from the cookbook. 

_One question:_ Should the cookbook just than link to the site, for the setup ? I think it would make sense to have one, centralized configuration page.

@abstractj++

**Comment by edewit** [comment]
@matzew right, after this PR is merged we can update the cookbooks to point to the site

**Comment by matzew** [comment]
awesome

On Wed, Feb 25, 2015 at 9:24 AM, Erik Jan de Wit <notifications@github.com>
wrote:

> @matzew <https://github.com/matzew> right, after this PR is merged we can
> update the cookbooks to point to the site
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear.org/pull/506#issuecomment-75922445>.
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

**Comment by corinnekrych** [comment]
https://issues.jboss.org/browse/AGIOS-403
https://issues.jboss.org/browse/AGCORDOVA-68
https://issues.jboss.org/browse/AGDROID-384
https://issues.jboss.org/browse/AGJS-286
Created for cookbook repos

**Comment by abstractj** [comment]
@edewit I will do the changes suggested by you today

@matzew Regarding the readme updates, that's was my plan after get some feedback here, anyways thanks @corinnekrych for creating those jiras. I created https://issues.jboss.org/browse/AGSEC-205 and attached the related Jiras there.

I will review all the comments here this morning and update this PR. Thanks in advance for the feedback.

**Comment by matzew** [comment]
cool beans! great seeing this initiative, @abstractj

On Wed, Feb 25, 2015 at 10:23 AM, Bruno Oliveira <notifications@github.com>
wrote:

> @edewit <https://github.com/edewit> I will do the changes suggested by
> you today
>
> @matzew <https://github.com/matzew> Regarding the readme updates, that's
> was my plan after get some feedback here, anyways thanks @corinnekrych
> <https://github.com/corinnekrych> for creating those jiras. I created
> https://issues.jboss.org/browse/AGSEC-205 and attached the related Jiras
> there.
>
> I will review all the comments here this morning and update this PR.
> Thanks in advance for the feedback.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear.org/pull/506#issuecomment-75929310>.
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

File: docs/guides/index.markdown [hunk 2]
```diff
@@ -3,22 +3,36 @@ layout: basic
...
 
+### Security
 
-#### Contributing to AeroGear
+* OTP
+  * [AeroGear Two-Factor Authentication with OTP](AeroGear-OTP)
+  * [AeroGear Two-Factor Authentication with OTP as Cordova Plugin](aerogear-cordova/AerogearCordovaOTP)
+
+* OAuth2
+  1. [Basic steps](security/oauth2-basic-steps)
```
Comment (lfryc) [inline docs/guides/index.markdown]: I would prefer something like `Getting Started with various OAuth2 Providers` (Basic Steps sounds too general imo)

File: docs/guides/index.markdown [hunk 3]
```diff
@@ -3,22 +3,36 @@ layout: basic
...
+  2. Client libraries
+
+     The following SDKs were crafted with the simplicity in mind and are constantly improving at each release.
```
Comment (lfryc) [inline docs/guides/index.markdown]: Shouldn't be OAuth2 client libs list part of Downloads page? Here I would like to see `Getting Started with OAuth2` (planned in [AGSEC-204](https://issues.jboss.org/browse/AGSEC-204)) that would eventually links to Download pages.

File: docs/guides/index.markdown (hunk 2 above)
Comment (abstractj) [inline docs/guides/index.markdown]: @lfryc that's the idea, be general, `various OAuth2 providers` is general in the same way. Most of the guides outside make use of this sentence.

File: docs/guides/index.markdown (hunk 3 above)
Comment (abstractj) [inline docs/guides/index.markdown]: @lfryc to me is kinda of annoying have to access each page like iOS page, Android page...and etc. Into this way if people want to see all the SDKs available, they are grouped in a single place.

This is not a replacement to the downloads page, at the same time doesn't hurt to group them altogether.

File: docs/guides/index.markdown (hunk 1 above)
Comment (abstractj) [inline docs/guides/index.markdown]: @matzew updated

**Comment by abstractj** [comment]
@edewit @corinnekrych updated

**Comment by edewit** [comment]
:+1:  like the blur you added on the google logged in use picture

**Comment by danielpassos** [comment]
:+1:

File: docs/guides/index.markdown (hunk 3 above)
Comment (lfryc) [inline docs/guides/index.markdown]: That's true, would it help to place that content on /security/ landing page?

File: docs/guides/index.markdown (hunk 2 above)
Comment (lfryc) [inline docs/guides/index.markdown]: What about `Getting Started with various OAuth2 Providers (Keycloak, Facebook, Google)`?

I just didn't expect what that link opened, but we may reiterate later on overall structure.

File: docs/guides/index.markdown (hunk 3 above)
Comment (abstractj) [inline docs/guides/index.markdown]: @lfryc I'm totally cool with it. I was wondering about craft the whole content at the guides section and later move to the landing page. Not sure what do you think, but move only few items, may look odd. Like a blank page with few items.

But I can do that for sure if you think it's better.

File: docs/guides/index.markdown (hunk 2 above)
Comment (abstractj) [inline docs/guides/index.markdown]: @lfryc We can always improve later, this is not something definitive. Also, with the new website we gonna have room enough to change. I think more important now is to see if the content looks good and later we can rearrange/move sections.

File: docs/guides/index.markdown (hunk 3 above)
Comment (lfryc) [inline docs/guides/index.markdown]: :+1: let's fit it into current structure, sorry I'm too new-site oriented atm ;-)

**Comment by abstractj** [comment]
Landed, thank you all

**Pull request closed without merging**